doc/track_2_migration_guidance.md

```diff
@@ -0,0 +1,100 @@
+# Track 2 Migration Guidance
+
+Azure CLI is built upon Azure Python SDK. Recently Azure Python SDK announced next generation product. It is named Track 2 SDK. The old version of SDK is named Track 1. It claims that it has great advantages than Track 1 SDK. It is not compatible with Track 1 SDK. Azure CLI developers need to spend considerable time and do some work to migrate from Track 1 SDK to Track 2 SDK.
+
+This documentation summarizes typical issues and solutions when adopting Track 2 SDK in Azure CLI.
+
+**Long running operation function name change**
+
+Long running operations have changed their function names in Track 2 SDK. A `begin_` prefix is added. E.g. `create_or_update` becomes `begin_create_or_update`. `delete` becomes `begin_delete`. It is a naming convention in Track 2 SDK to indicate that an operation is a long running operation.
+
+**Property name change**
+
+Some of property names change in Track 2 SDK.
+
+Examples:
+
+```
+hyperVgeneration -> hyperVGeneration
+disk_mbps_read_only -> disk_m_bps_read_only
+disk_mbps_read_write -> disk_m_bps_read_write
+virtual_machine_extension_type -> type_properties_type
+type1 -> type_properties_type
+instance_ids -> vm_instance_i_ds
+diskMbpsReadWrite -> diskMBpsReadWrite
+```
+
+Some changes are unnecessary, even wrong in English.
```

Comment:
not necessary

Reply:
Above changes should be fixed right? Do we have issue to track this?

Reply:
> 
> 
> not necessary

why
https://dictionary.cambridge.org/us/dictionary/english/unnecessary

Reply:
> 
> 
> Above changes should be fixed right? Do we have issue to track this?

Yes. Yes. I opened one. https://github.com/Azure/autorest.python/issues/850